src/backend/utils/adt/lockfuncs.c

```diff
@@ -84,6 +79,88 @@ VXIDGetDatum(BackendId bid, LocalTransactionId lxid)
 	return CStringGetTextDatum(vxidstr);
 }
 
+/*
+ * This query relies on the existance of pg_locks view to use a force random
+ * distribution on it.
+ */
+#define PG_LOCKS_INTERNAL_QUERY \
+	"select * from gp_dist_random('pg_catalog.pg_locks')"
+
+/*
+ * Build a querydesc for SELECT on pg_locks relation to be passed to executor.
+ */
+static QueryDesc *
+build_pg_locks_querydesc(void)
```

Comment:
```suggestion
build_pg_locks_querydesc(const char *sql)
```
so that the compiler does not copy the string from the `PG_LOCKS_INTERNAL_QUERY` macro THREE times

Reply:
https://en.wikipedia.org/wiki/Common_subexpression_elimination